### PR TITLE
[UNTIL-9449] Added gradle catalogue

### PR DESCRIPTION
**Pull request by shekar-allam**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -49,27 +35,34 @@ android {
...
     // Module Specific dependencies
-    implementation(Dependencies.AndroidX.CONSTRAINT_LAYOUT)
+    implementation(libs.androidx.constraintlayout)
 
     // ML Kit
-    api(Dependencies.Google.MLKIT_BARCODES)
-    api(Dependencies.CameraX.CAMERA2)
-    api(Dependencies.CameraX.CAMERA_LIFECYCLE)
-    api(Dependencies.CameraX.CAMERA_VIEW)
+    api(libs.bundles.mlkit)
```
Comment (djordjeh) [inline build.gradle.kts:57]: change to implementation

File: build.gradle.kts [hunk 2]
```diff
@@ -49,27 +35,34 @@ android {
...
+
+    // Camera
+    api(libs.bundles.camera)
```
Comment (djordjeh) [inline build.gradle.kts:60]: same as above

File: src/main/java/de/tillhub/scanengine/Scanner.kt [hunk 3]
```diff
@@ -1,69 +1,20 @@
...
-     * Can be used to observe if scanning is in progress.
-     */
-    fun scanningInProgress(): Flow<Boolean>
+    fun observeScannerResults(): Flow<ScanEvent>
 
     /**
      * Start the camera based scanner.
      */
-    fun scanCameraCode(scanMultipleCodes: Boolean = false, scanKey: String? = null)
+    fun scanCameraCode(scanKey: String?)
```
Comment (djordjeh) [inline src/main/java/de/tillhub/scanengine/Scanner.kt:17]: startCameraScanner

File: src/main/java/de/tillhub/scanengine/common/CoroutineScopeProviderImpl.kt [hunk 4]
```diff
@@ -3,8 +3,7 @@ package de.tillhub.scanengine.common
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.SupervisorJob
-import javax.inject.Inject
 
-class CoroutineScopeProviderImpl @Inject constructor() : CoroutineScopeProvider {
+class CoroutineScopeProviderImpl : CoroutineScopeProvider {
```
Comment (djordjeh) [inline src/main/java/de/tillhub/scanengine/common/CoroutineScopeProviderImpl.kt:7]: Do we need this anymore?

File: src/main/java/de/tillhub/scanengine/google/GoogleScanner.kt [hunk 5]
```diff
@@ -1,98 +1,57 @@
...
-import de.tillhub.scanengine.ScannerConnection
 import de.tillhub.scanengine.google.ui.GoogleScanningActivity
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.MutableStateFlow
-import timber.log.Timber
-import javax.inject.Singleton
+import java.lang.IllegalStateException
+import java.lang.ref.WeakReference
+
+class GoogleScanner(
```
Comment (djordjeh) [inline src/main/java/de/tillhub/scanengine/google/GoogleScanner.kt:20]: Lets rename this to GeneralScanner or DefaultScanner

**Review by djordjeh — APPROVED** [review]

**Review by SloInfinity — APPROVED** [review]

**Review by djordjeh — APPROVED** [review]

**Pull request merged**